pyproject.toml

```diff
@@ -53,6 +53,45 @@ boto3 = [
 Repository = "https://github.com/aio-libs/aiobotocore"
 Documentation = "https://aiobotocore.aio-libs.org"
 
+[dependency-groups]
+botocore-dev = [
+    # keep in sync with https://github.com/boto/botocore/blob/develop/requirements-dev.txt
+    # "wheel==0.43.0",
+    # "behave==1.2.5",
+    # "jsonschema==4.21.1",
+    "coverage==7.2.7",
+    # "setuptools==71.1.0;python_version>='3.12'",
+    # "packaging==24.1;python_version>='3.12'", # Requirement for setuptools>=71
+
+    # Pytest specific deps
+    "pytest==8.1.1",
+    "pytest-cov==5.0.0",
+    "pytest-xdist==3.5.0",
+    # "atomicwrites>=1.0", # Windows requirement
+    # "colorama>0.3.0", # Windows requirement
+]
+dev = [
+    "dill ~=0.3.3", # Used in test_patches.py
+    "docker ~=7.1",
+    "docutils ~=0.16", # Used in test_version.py
+    "moto[server,s3,sqs,awslambda,dynamodb,cloudformation,sns,batch,ec2,rds] ~=4.2.9",
+    "packaging ~=24.1", # Used in test_version.py
+    "pip ~=24.3.1", # Used in test_version.py
+    "pre-commit ~=3.5.0",
+    "pytest-asyncio ~=0.23.8",
+    "requests ~=2.32.3", # Used in test_version.py
+    "time-machine ~=2.15.0",
```

Comment:
this is new to me, needed?

Reply:
Again, these are the exact same dev dependencies that we already use and which have been specified in `dependencies-dev.in`. This is now migrated to [dependency-groups] in `pyproject.toml`.

Reply:
please don't resolve conversations started by reviewers, bad etiquette.  Anywho this seems like it was added in https://github.com/aio-libs/aiobotocore/commit/a73f8c26192b190f503791b309124cb32eac44c8 for seemingly no reason.  We should not be adding random requirements that aren't needed

Reply:
if it is needed it needs a comment as to why it's here

Reply:
> please don't resolve conversations started by reviewers, bad etiquette.

Sorry, I was not aware and will not do so again.

> Anywho this seems like it was added in [a73f8c2](https://github.com/aio-libs/aiobotocore/commit/a73f8c26192b190f503791b309124cb32eac44c8) for seemingly no reason. We should not be adding random requirements that aren't needed

It's a valid requirement: https://github.com/aio-libs/aiobotocore/blob/74323389b8fb6fcbe415f27765e7d7712f440050/tests/boto_tests/unit/test_signers.py#L75

> if it is needed it needs a comment as to why it's here

OK, I will add that comment referring to `test_signers.py`. My hesitation is due to the fact that those cross references do not scale well, i.e. for dev dependencies that are used in multiple or many modules.

Reply:
The comment is now included in this PR.

Reply:
ok cool, thanks for that!